### ISSUE TITLE
[1.4.1] Error: ENOENT: no such file or directory, open '/tmp/mt-1jpixmp/os.json'

### ISSUE THREAD

**Issue by lyquocnam** [meteor/meteor#7666]
error when create new project
`meteor create test --release 1.4.1`
```
/home/lynam/.meteor/packages/meteor-tool/.1.4.1.15qh84w++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/meteor-promise/promise_server.js:165
      throw error;
      ^

Error: ENOENT: no such file or directory, open '/tmp/mt-1jpixmp/os.json'
    at Error (native)

```

**Comment by benjamn** [comment]
Does this still happen after running `meteor update` to update to Meteor 1.4.1?

**Comment by benjamn** [comment]
Isn't this a duplicate of https://github.com/meteor/meteor/issues/7657?

**Comment by alfiepoleon** [comment]
So I ran meteor upgrade and got this error.

```
> ❯ meteor update
> /Users/alfred/.meteor/packages/meteor-tool/.1.4.1.1ug1jjz++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/isopackets/ddp/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:165
>       throw error;
>       ^
> 
> Error: ENOENT: no such file or directory, open '/private/var/folders/04/l64f_q8x6y3d8hqr3jh5zt9r0000gp/T/mt-u2wovu/plugin.compileStaticHtmlBatch.os/program.json'
>     at Error (native)
```
Upgrading from 1.4.0.1

**Comment by TheDocBrown** [comment]
I attempted deploying Telescope from scratch and received this error message after attempting to initiate Meteor. After realizing I'd attempted to run Meteor with Nodejs 6.4.0 as the basis, I reverted my install to Nodejs 4.5.0 LTS, and received the same message. Are there any other prerequisites, aside from the aforementioned additional tools which I've also installed?

**Comment by ahmadio** [comment]
I faced the same issue while creating new meteor project, I found this solution to fix it.

**Comment by ahmadio** [comment]
I faced the same issue while creating new meteor project, I found this solution to fix it.
[stackoverflow](http://stackoverflow.com/questions/39153477/how-do-i-resolve-this-error-enoent-no-such-file-or-file-or-directory-open-t)

**Comment by schankam** [comment]
Thanks @ahmadio , it worked for me...

**Comment by matiasherranz** [comment]
Thanks @ahmadio, worked for me.

**Comment by smeijer** [comment]
the `/etc/hosts` solution doesn't work on Windows. I'm having a huge problem now, because I'm unable to work on any of my projects.

**Comment by abernix** [comment]
For Windows users, I would suggest using the much more simple workaround I identified in this comment instead: https://github.com/meteor/meteor/issues/7766#issuecomment-254445923

Unix users who prefer not to modify their `/etc/hosts` can also check here: https://github.com/meteor/meteor/issues/7806#issuecomment-254125662

/cc @smeijer

**Comment by smeijer** [comment]
@abernix that solution doesn't work either. I guess it does the same as changing the host file? (on windows: `c:\windows\system32\drivers\etc\hosts`

**Comment by abernix** [comment]
@smeijer Does this happen with the official, released version?  Or just the checked-out version?

I would guess this has to do with some bugs that exist when using packages with colons (`:`) in their name (as the character is problematic on Windows).  Is that `meteorhacks:unblock` package in your own `packages` directory?  If so, what is it called?

**Comment by smeijer** [comment]
@abernix; It happend with the official released version as well as the checked-out version.

I cleaned up my temp directories, including `%localappdata%\temp\`, and removed some old meteor tools from `%localappdata%\.meteor\packages\meteor-tool`. As I'm using meteor 1.3+ in my projects, it felt safe to delete everything under `1.3.x`.

Cleaning the temp directory seemed to be fatal to me. Meteor did not want to redownload packages that it was missing. Sometimes complaining about missing `meteorhacks:unblock`, sometimes complaining about `aldeed:..`. All referring to missing `temp\mt-xxxxx` directories. I don't believe it's the package that had something wrong, as it always had worked, and it is working now again.

It did not even help to completely remove meteor, and install it again. Even when creating a new project, `meteor add meteorhacks:unblock` failed. At the end, I got meteor working again by using an offline installer for meteor 1.1, that I found somewhere on the forums.

I can hear your big question; why cleanup the temp directories!?!?!, Well. Because `meteor` was using 15gb from my SSD. Now I have everything working again, it's only 2gb (and yesterday it was even smaller), including the 1.1 installation I don't need. It would be really awesome if we could just drop all packages that are not being used anymore.

I mean; come on. I have this installation 2 days, and it already contains 4 versions of packages that I cannot get rid of.

![image](https://cloud.githubusercontent.com/assets/1196524/19552209/2da2c528-96af-11e6-9f81-9ae2d6f699d6.png)

**Comment by abernix** [comment]
@smeijer I agree that this the space consumption can get overwhelming, especially if you're moving between versions frequently.  Meteor does keep cached versions of packages around.  Unfortunately, as these are global caches and its difficult to do anything automatically as you could have any number of Meteor projects spread anywhere across your computer that would need to be found, scanned and unused dependencies identified.  Right now #6580 is probably a reasonable step in the right direction.  For now, once you've updated all your projects to the latest Meteor (and its packages too), a fresh reinstall (which on Windows would involve removing your `%LocalAppData/.meteor/` directory) would be the best advice.  Anyhow, this is off  topic for this issue so further discussion should take place in another issue (perhaps the one I linked).
